.jenkins/Jenkinsfile

```diff
@@ -0,0 +1,93 @@
+//The Jenkinsfile runs entire build in a golang container and mounts the /var/run/docker.sock file to allow access to the host docker within the container
+node('wcm') {
+    def build_ok = true
+    docker.image('golang:1.15-buster').inside('-u root --net=host -v /var/run/docker.sock:/var/run/docker.sock') {
```

Comment:
we'll possibly have an issue with using `-u root` in the long run.  Got some feedback on that from the SRE team that it could be causing content to be left on the jenkins slave that the container runs on.

Reply:
Ok, but I can leave the `-u root` there for now?

Reply:
If you could try without `-u root` to see if it's still working the same that'd be good.  To get kind cluster create and go get / go mod stuff to run in nsm-nse I had to add the following to env vars:
```
        HOME        = "${WORKSPACE}"
        GOCACHE     = "/go/.cache"
```

Reply:
Sure, I can give that a go and see what happens.

Reply:
This is what it looks like without `-u root`: https://engci-private-sjc.cisco.com/jenkins/eti-sre/job/AppN/job/amimam/job/networkservicemesh/job/devop_jenkins/143/console

Reply:
Ideally, we would have access to an ephemeral docker host, or the build agent would be ephemeral, so that we can easily guarantee that our environments are consistent. There's too many hands touching those build agents to have any sort of guarantee that they're going to be in a consistent state.